pages/api/admin/index.ts

```diff
@@ -24,8 +25,59 @@ export default async function handler(
           error: 'Sorry, an error occurred while connecting to the database',
         });
       }
-      break;
 
+    case 'POST':
+      try {
+        // start a try catch block to catch any errors in parsing the request body
+        const admin = req.body as QueriedAdminData;
+
+        console.log('Here is the request body: ', admin);
+
+        // Get the user's email and password from the request body
+        const { firstName, lastName, email, password } = admin;
+
+        // Check if the user's email and password are valid
+        if (!email || !email.includes('@') || !password) {
+          res.status(422).json({ message: 'Invalid email or password' });
+          throw new Error('Invalid email or password');
+        }
+
+        // Connect to the database
+        await dbConnect();
+
+        // Check if the user already exists in database
+        const checkExisting = (await Admins.findOne({
+          email,
+        })) as QueriedAdminData;
+
+        // If the user already exists, return an error
+        if (checkExisting) {
+          res.status(422).json({ message: 'Admin email already exists' });
+          throw new Error('Admin email already exists');
+        }
```

Comment:
Before you insert the user, can you first check if the current user has an admin status?